openbas-model/src/main/java/io/openbas/database/model/InjectTestStatus.java

```diff
@@ -0,0 +1,161 @@
+package io.openbas.database.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.openbas.database.converter.InjectStatusExecutionConverter;
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.hibernate.annotations.UuidGenerator;
+
+import java.time.Instant;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Setter
+@Getter
+@Entity
+@Table(name = "injects_tests_statuses")
+public class InjectTestStatus implements Base {
+
+  @Id
+  @Column(name = "status_id")
+  @GeneratedValue(generator = "UUID")
+  @UuidGenerator
+  @JsonProperty("status_id")
+  private String id;
+
+  @Column(name = "status_name")
+  @JsonProperty("status_name")
+  @Enumerated(EnumType.STRING)
+  @NotNull
+  private ExecutionStatus name;
+
+  // region dates tracking
+  @Column(name = "status_executions")
+  @Convert(converter = InjectStatusExecutionConverter.class)
+  @JsonProperty("status_traces")
+  private List<InjectStatusExecution> traces = new ArrayList<>();
+
+  @Column(name = "tracking_sent_date")
+  @JsonProperty("tracking_sent_date")
+  private Instant trackingSentDate; // To Queue / processing engine
+
+  @Column(name = "tracking_ack_date")
+  @JsonProperty("tracking_ack_date")
+  private Instant trackingAckDate; // Ack from remote injector
+
+  @Column(name = "tracking_end_date")
+  @JsonProperty("tracking_end_date")
+  private Instant trackingEndDate; // Done task from injector
+
+  @Column(name = "tracking_total_execution_time")
+  @JsonProperty("tracking_total_execution_time")
+  private Long trackingTotalExecutionTime;
+  // endregion
+
+  // region count
+  @Column(name = "tracking_total_count")
+  @JsonProperty("tracking_total_count")
+  private Integer trackingTotalCount;
+
+  @Column(name = "tracking_total_error")
+  @JsonProperty("tracking_total_error")
+  private Integer trackingTotalError;
+
+  @Column(name = "tracking_total_success")
+  @JsonProperty("tracking_total_success")
+  private Integer trackingTotalSuccess;
+  // endregion
+
+  @OneToOne
+  @JoinColumn(name = "status_inject_inject_id")
+  @JsonIgnore
+  private Inject inject;
+
+  @JsonProperty("inject_title")
+  public String getInjectTitle() {
+    return inject.getTitle();
+  }
+
+  @JsonProperty("injector_contract")
+  public Optional<InjectorContract> getInjectContract() {
+    return inject.getInjectorContract();
+  }
```

Comment:
Do we need the whole contract?

Reply:
Yes, it is necessary to use the InjectIcon component